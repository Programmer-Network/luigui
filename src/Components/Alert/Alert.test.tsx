import { render, screen } from "@testing-library/react";
import { Alert } from "index";

describe("Alert component", () => {
  it("renders correctly with children", () => {
    const { asFragment } = render(
      <Alert type='success'>Success Message</Alert>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("applies correct class for success type", () => {
    render(<Alert type='success'>Success Message</Alert>);
    expect(screen.getByText("Success Message")).toHaveClass(
      "border-green-400",
      "bg-green-300",
      "text-green-700"
    );
  });

  it("applies correct class for error type", () => {
    render(<Alert type='error'>Error Message</Alert>);
    expect(screen.getByText("Error Message")).toHaveClass(
      "border-red-400",
      "bg-red-300",
      "text-red-700"
    );
  });

  it("applies correct class for warning type", () => {
    render(<Alert type='warning'>Warning Message</Alert>);
    expect(screen.getByText("Warning Message")).toHaveClass(
      "border-yellow-400",
      "bg-yellow-300",
      "text-yellow-700"
    );
  });

  it("applies correct class for info type", () => {
    render(<Alert type='info'>Info Message</Alert>);
    expect(screen.getByText("Info Message")).toHaveClass(
      "border-blue-400",
      "bg-blue-300",
      "text-blue-700"
    );
  });

  it("allows custom class names to be applied", () => {
    render(
      <Alert type='success' className='custom-class'>
        Success Message
      </Alert>
    );
    expect(screen.getByText("Success Message")).toHaveClass("custom-class");
  });
});
