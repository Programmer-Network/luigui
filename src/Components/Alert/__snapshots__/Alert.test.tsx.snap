// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert component > renders correctly with children 1`] = `
<DocumentFragment>
  <div
    class="flex min-h-screen max-w-screen-sm items-center justify-center"
  >
    <div
      class="bg-gradient-to-r p-[1px] from-green-400 via-green-500 to-green-900 text-green-500"
    >
      <div
        class="back flex h-full w-full items-center justify-center bg-primary-background-color p-4"
      >
        <span>
          Success Message
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
